### DISCUSSION TITLE
Can I create a window without an even loop?

### DISCUSSION THREAD

**Discussion by psionic12** [rust-windowing/winit#2708] [Q&A]
Hi, In my case I need a `HasRawWindowHandle` to call some graphic APIs (for example, what surface this machine supports), and I don't really need an event loop. But I found that the only way to create a window is to pass a `EvenLoop` to the `build` function.

So can I get a window, or in this case, another type that implements `HasRawWindowHandle`, or a `RawWindowHandle` value, by using winit, without creating an event loop? If it is impossible, is there any other way to do this?

**Comment by kchibisov** [comment]
It's impossible. You need connection to the display server to communicate with it and handle window events, which is an event loop in particular, like the entire reason for event loop passed to window, is because you need event loop to create window on a level that is below the winit API.

**Reply by psionic12** [reply]
Then, can I somehow delete an even loop? It seems that simply drop it will not actually destroy it. The root problem here is that there's another part which actually do the even handling things, but this lead to two even_loops exists.

**Reply by kchibisov** [reply]
What platform are you on? I guess macOS/Windows? Just create a window where you event loop is. It just seems you don't quite understand what you're doing. If you want headless rendering surfaces, you don't need winit or windows at all.

**Reply by madsmtm** [reply]
I think what may have you confused is that this _would_ be possible on some platforms like macOS where when you create a window, the system keeps track of it, while I don't think it is on e.g. Wayland, where you have to keep track of the window yourself

**Reply by kchibisov** [reply]
Yeah, on Wayland/X11 you must have a connection(event_loop) for your window.

**Reply by psionic12** [reply]
Hi, I know what I'm doing, It just that I may not make myself clear: I used a crate which internally create an event loop for window event handling, and the event loop is not exposed. On the other hand, I need a window to call some graphic API (and I'm not doing surfaceless rendering), in the case, the program will crash due to create two event loops. That's why I asked. You may say the crate I'm using which do not expose event loop makes no sense, but here is where I am. 

If you are asking what graphic API do I call that need a window: I need a window to get a surface to get information like how many queue families, how many images in a swapchain, and so on. Just information, no rendering at all. 

This may explain why I want a window, and why I want to successfully drop an event loop after I get what I want (currently drop an event loop not works).

Hope I make myself clear, and thank you for the help.

**Reply by kchibisov** [reply]
>create an event loop for window event handling, and the event loop is not exposed

Can this crate create a window for you then? I'm not sure why you need winit here. Just create a window is not how it works in general. On most platforms dropping event loop will bring down everything as well.

On Linux dropping will drop graphics API all together for example, because they require event loop handle(aka wl_display, x11 display).

So no, winit can't have windows without event loop, because it's simply illogical and useless in general case.

**Reply by psionic12** [reply]
Okay, thanks for your help. And thanks for the knowledge of how window server works.
